api/v1beta1/proposal_webhook.go

```diff
@@ -259,3 +264,21 @@ func validateDeleteFedMember(ctx context.Context, c client.Client, deleteMember,
 	}
 	return nil
 }
+
+func AddEndAtAnnotation(r client.Reader, p Proposal) (*Federation, error) {
+	fed := &Federation{}
```

Comment:
we should check whether this label exists. If yes, we should block this proposal creation.

Reply:
https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks
According to the documentation, **the mutatingwebhook will be called first**, where the relevant annotation has been added to the federation, and the logic is incorrect when judged by the validatingwebhook.

If we want to do validation on the creation of a proposal for a federation type, then the logic for adding annotations should not be done at the wehbook level.